.github/workflows/run-tests.yml

```diff
@@ -20,7 +15,40 @@ on:
         default: ${{ github.sha }}
 
 jobs:
-  do-nothing:
-    runs-on: ubuntu-latest
+  unit-tests:
+    strategy:
+      # By default, if any job in a matrix fails, all other jobs are immediately canceled. This makes the jobs run
+      # to completion instead.
+      fail-fast: false
+      matrix:
+        node: ${{ fromJson(inputs.node-matrix) }}
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
     steps:
-      - run: echo DOING NOTHING FOR NOW
+      # Check out the code
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.target-branch }}
+      # Make sure we're on the right version on Node
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node.version }}
+      - run: yarn
+      - run: yarn build
+      - run: yarn test
+        id: jest-tests
+      - run: yarn lint
+      - if: ${{ failure() && steps.jest-tests.outcome != 'success' }}
+        # Use bash for this to avoid any syntax nonsense
+        shell: bash
+        run: |
+          MSG="Jest tests failed. If no annotations were added, the most likely problem is insufficient code coverage."
+          echo "::error ::$MSG"
+      - name: Upload full artifact
```

Comment:
Just curious... have we ever used in the past the code coverage artifacts that are uploaded? I can easily just run the report locally if needed. This is fine - but I wonder if it is ever used.

Reply:
I have never used this. +1 to removing this if no one else uses.

Reply:
I've personally used the unit test artifact because it's helpful when a test fails on one system but not another. And at least in the Java world we saw cases where Windows and UNIX calculated coverage _slightly_ differently. That might not be a concern for TypeScript.